homeassistant/components/camera/__init__.py

```diff
@@ -273,7 +273,7 @@ async def preload_stream(hass, _):
 
             request_stream(hass, source, keepalive=True, options=camera.stream_options)
 
-    async_when_setup(hass, DOMAIN_STREAM, preload_stream)
+    hass.bus.async_listen_once(EVENT_HOMEASSISTANT_START, preload_stream)
```

Comment:
Why isn't the old approach good? I'm just curious.

Reply:
It's a bug. It needs to iterate over the entities. When `stream` is set up it's not ready with adding entities. So listening to `START` at least guarantees all platforms have set up their entities. It's still not perfect as camera entities might be loaded after start that also have preload stream activated…